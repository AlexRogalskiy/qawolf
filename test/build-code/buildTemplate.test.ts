import {
  buildScriptTemplate,
  buildTestTemplate,
} from '../../src/build-code/buildTemplate';

describe('buildTemplate', () => {
  it('builds script template', () => {
    const template = buildScriptTemplate({
      name: 'myScript',
      url: 'www.qawolf.com',
    });
    expect(template).toMatchSnapshot();
  });

  it('builds test template', () => {
    const template = buildTestTemplate({
      name: 'myTest',
      url: 'www.qawolf.com',
    });
    expect(template).toMatchSnapshot();
  });

  it('throws an error if device does not exist', () => {
    const testFn = (): string =>
      buildTestTemplate({
        device: 'unknown',
        name: 'myTest',
        url: 'www.qawolf.com',
      });
    expect(testFn).toThrowError('Device unknown not available in Playwright');
  });

  it('includes device if specified', () => {
    const template = buildScriptTemplate({
      device: 'iPhone 11',
      name: 'myScript',
      url: 'www.qawolf.com',
    });
    expect(template).toMatchSnapshot();

    const template2 = buildTestTemplate({
      device: 'iPad Mini',
      name: 'myTest',
      url: 'www.qawolf.com',
    });
    expect(template2).toMatchSnapshot();
  });
});
