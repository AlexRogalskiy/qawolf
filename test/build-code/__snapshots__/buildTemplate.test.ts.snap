// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`buildTemplate builds script template 1`] = `
"const qawolf = require(\\"qawolf\\");

const myScript = async context => {
  let page = await context.newPage();
  await page.goto(\\"www.qawolf.com\\");
  await qawolf.create();
};

exports.myScript = myScript;

if (require.main === module) {
  (async () => {
    const browser = await qawolf.launch();
    const context = await browser.newContext();
    await myScript(context);
    await browser.close();
  })();
}"
`;

exports[`buildTemplate builds test template 1`] = `
"const qawolf = require(\\"qawolf\\");

let browser;
let page;

beforeAll(async () => {
  browser = await qawolf.launch();
  const context = await browser.newContext();
  page = await context.newPage();
});

afterAll(() => browser.close());

test('myTest', async () => {
  await page.goto(\\"www.qawolf.com\\");
  await qawolf.create();
});"
`;

exports[`buildTemplate includes device if specified 1`] = `
"const { devices } = require(\\"playwright\\");
const qawolf = require(\\"qawolf\\");
const device = devices[\\"iPhone 11\\"];

const myScript = async context => {
  let page = await context.newPage();
  await page.goto(\\"www.qawolf.com\\");
  await qawolf.create();
};

exports.myScript = myScript;

if (require.main === module) {
  (async () => {
    const browser = await qawolf.launch();
    const context = await browser.newContext({
      userAgent: device.userAgent,
      viewport: device.viewport
    });
    await myScript(context);
    await browser.close();
  })();
}"
`;

exports[`buildTemplate includes device if specified 2`] = `
"const { devices } = require(\\"playwright\\");
const qawolf = require(\\"qawolf\\");
const device = devices[\\"iPad Mini\\"];

let browser;
let page;

beforeAll(async () => {
  browser = await qawolf.launch();
  const context = await browser.newContext({
      userAgent: device.userAgent,
      viewport: device.viewport
    });
  page = await context.newPage();
});

afterAll(() => browser.close());

test('myTest', async () => {
  await page.goto(\\"www.qawolf.com\\");
  await qawolf.create();
});"
`;
